elastic/elastic-agent
# Fix timeout in coordinator shutdown loop

## What does this PR do?

Fixes https://github.com/elastic/elastic-agent/issues/2394 

This does two things:
- Fixes a bug where if the coordinator shutdown timed out, we wouldn't break from the loop
- Improves the shutdown process, we'll now shutdown without a timeout in cases where subcomponents return a `nil` error.

## Why is it important?

This is a bug

## Checklist


## How to test this PR locally

- Pull and build
- Run the agent, enroll in fleet
- from the terminal, send the agent process a SIGINT or SIGTERM
- The agent should shut down after 1-10 seconds.

## Related issues

- fixes https://github.com/elastic/elastic-agent/issues/2391
- fixes https://github.com/elastic/elastic-agent/issues/2392
